src/cmd/ksh93/bltins/typeset.c

```diff
@@ -1090,9 +1092,9 @@ int	b_builtin(int argc,char *argv[],Shbltin_t *context)
 	struct tdata tdata;
 	Shbltin_f addr;
 	Stk_t	*stkp;
-	void *library=0;
 	char *errmsg;
 #if SHOPT_DYNAMIC
+	void *library=0;
 	unsigned long ver;
 	int list = 0;
 	char path[1024];
```

Comment:
The `ver` variable and the `path` array are only used in the `if(arg)` block on lines 1159-1169, so it's probably better to define them within that block. Plus, the `1024` should be `PATH_MAX`.